analyzer/java/com/engflow/bazel/invocation/analyzer/bazelprofile/BazelProfile.java

```diff
@@ -204,6 +205,18 @@ public ImmutableMap<String, String> getOtherData() {
     return ImmutableMap.copyOf(otherData);
   }
 
+  /**
+   * For performance reasons, prefer getting the Datum {@link BazelVersion} provided by {@link
```

Comment:
Why even expose this from this class here? From what I can see the only use of this from this class is in the summary.

Reply:
#115 gives an example where the logic within `BazelProfile` uses the version. I'm not sure we'll need it anymore, but in principle it does make sense that the `BazelProfile` code might have to look into different places for the functionality it exposes. In particular, for returning the main thread, critical path, GC thread and action counts it may be necessary to look into different places of the JSON depending on the Bazel version that generated the profile.

Reply:
I've adjusted the code so it doesn't matter whether you fetch the `BazelVersion` through the `BazelProfile` directly or through the `DataManager` (via `BazelVersionDataProvider`). For convenience, I still think the separate `DataProvider` makes sense.